cl/search/tasks.py

```diff
@@ -1681,3 +1647,113 @@ def remove_documents_by_query(
         es_document._index.refresh()
 
     return response
+
+
+@app.task(
+    bind=True,
+    autoretry_for=(
+        botocore_exception.HTTPClientError,
+        botocore_exception.ConnectionError,
+    ),
+    max_retries=5,
+    retry_backoff=10,
+)
+def retrieve_embeddings(
+    self,
+    opinion_ids: list[int],
+    directory: str = "opinions",
+) -> list[dict] | None:
+    """Download opinion embeddings from S3 for the given opinion IDs.
+
+    :param self: The Celery task instance.
+    :param opinion_ids: A list of opinion IDs for which to retrieve embeddings.
+    :param directory: The directory where the embeddings are stored.
+    :return: A list of dictionaries containing the embeddings.
+    """
+
+    storage = AWSMediaStorage()
+
+    def download_embedding(opinion_id: int) -> dict | None:
+        file_path = os.path.join(
```

Comment:
I feel bad saying this, but this might be that rare time when os.path.join is the wrong tool. If this was run on a Windows machine, it'd use a `\` to join the path, which wouldn't work. I think the better way to do this is `'/'.join()` even though it hurts.

Reply:
Good point, thanks! I did some research and it seems that we can use `pathlib.PurePosixPath` to make the S3 path generation OS agnostic. I've gone ahead and applied the tweak. I also applied the same tweak in:
https://github.com/freelawproject/courtlistener/pull/5046